File: java/src/utils/Character.java
package utils;

import java.util.ArrayList;
import java.util.HashMap;

public class Character {


    private int id;
    private String name;
    private int health;
    private ArrayList<Integer> location;
    private ArrayList<Integer> base;

    private int spawn;
    private int points;
    private int carrying;

    private String status;

    public Character(int id, String name, int health, ArrayList<Integer> location, ArrayList<Integer> base, int spawn, int points, int carrying, String status) {
        this.id = id;
        this.name = name;
        this.health = health;
        this.location = location;
        this.base = base;
        this.spawn = spawn;
        this.points = points;
        this.carrying = carrying;
        this.status = status;
    }

    public static Character parse(HashMap<String, Object> character_state){
        int spawn = (int) character_state.get("spawn");
        String name = (String) character_state.get("name");
        int health = (int) character_state.get("health");
        ArrayList<Integer> location = (ArrayList<Integer>) character_state.get("location");
        int id = (int) character_state.get("id");
        int points = (int) character_state.get("points");
        int carrying = (int) character_state.get("carrying");
        ArrayList<Integer> base = (ArrayList<Integer>) character_state.get("base");
        String status = (String) character_state.get("status");
        return new Character(id, name, health, location, base, spawn, points, carrying, status);
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getHealth() {
        return health;
    }

    public ArrayList<Integer> getLocation() {
        return location;
    }

    public ArrayList<Integer> getBase() {
        return base;
    }

    public int getSpawn() {
        return spawn;
    }

    public int getPoints() {
        return points;
    }

    public int getCarrying() {
        return carrying;
    }

    public String getStatus() {
        return status;
    }
}

File: src/game/Character.py

class Character:

    MAX_HEALTH = 100

    def __init__(self, character_id):
        self.id = character_id
        self.health = self.MAX_HEALTH
        self.points = 0
        self.carrying = 0
        self.location = (0, 0)
        self.base = (0, 0)
        self.status = 'alive'
        self.spawn_counter = 0
        self.name = 'Default'

    def damage(self, amount):
        if self.is_alive():
            if self.location != self.base:
                self.health -= amount

            if self.health <= 0:
                lost = self.carrying
                self.carrying = 0
                self.status = 'dead'
                self.spawn_counter = 10
                self.location = self.base
                return lost

        return 0

    def heal(self):
        self.health += 10
        if self.health > self.MAX_HEALTH:
            self.health = self.MAX_HEALTH

    def store(self):
        if self.location == self.base:
            self.points += self.carrying
            self.carrying = 0

    def is_dead(self):
        return self.status == 'dead'

    def is_alive(self):
        return self.status == 'alive'

    def get_location(self):
        return self.location

    def get_id(self):
        return self.id

    def respawn(self):
        self.health = self.MAX_HEALTH
        self.location = self.base
        self.status = 'alive'

    def disconnect(self):
        self.location = self.base
        self.status = 'disconnected'

    def __str__(self):
        stats = {
            'id': str(self.id),
            'health': str(self.health),
            'carrying': str(self.carrying),
            'points': str(self.points),
            'location': str(self.location),
            'base': str(self.base),
            'status': self.status,
            'spawn': str(self.spawn_counter),
            'name': self.name
        }
        return str(stats)
